### PR TITLE
DP-1256 - Create Orc examples on GitHub

### PR DESCRIPTION
**Pull request by gvjoshi25**

### CODE REVIEW

File: src/main/java/com/northconcepts/datapipeline/examples/amazons3/ReadAnOrcFileFromAmazonS3UsingATemporaryFile.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                int lengthRead;
+                while ((lengthRead = in.read(buffer)) > 0) {
+                    out.write(buffer, 0, lengthRead);
+                    out.flush();
+                }
+
+                DataReader reader = new OrcDataReader(orcFile);
+                DataWriter writer = new StreamWriter(System.out);
+
+                Job.run(reader, writer);
```
Comment (DeleTaylor) [inline src/main/java/com/northconcepts/datapipeline/examples/amazons3/ReadAnOrcFileFromAmazonS3UsingATemporaryFile.java]: Any reason this shouldn't happen after we've closed the AWS connection below?  Let's separate downloading the temp file from processing it if we can.

File: src/main/java/com/northconcepts/datapipeline/examples/amazons3/WriteAnOrcFileToAmazonS3UsingATemporaryFile.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        try {
+            File orcFile = File.createTempFile("credit-balance", ".orc");
+            orcFile.deleteOnExit();
+
+            try {
+                DataReader reader = new CSVReader(new File("example/data/input/credit-balance.csv"))
+                        .setFieldNamesInFirstRow(true);
+                OrcDataWriter writer = new OrcDataWriter(orcFile);
+
+                Job.run(reader, writer);
```
Comment (DeleTaylor) [inline src/main/java/com/northconcepts/datapipeline/examples/amazons3/WriteAnOrcFileToAmazonS3UsingATemporaryFile.java]: Similar to above, can we do this before opening the S3 connection?

File: src/main/java/com/northconcepts/datapipeline/examples/orc/ReadSelectedFieldsFromAnOrcFile.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import com.northconcepts.datapipeline.job.Job;
+import com.northconcepts.datapipeline.orc.OrcDataReader;
+
+public class ReadSelectedFieldsFromAnOrcFile {
+
+    public static void main(String[] args) {
+        FieldList columns = new FieldList("list", "double1", "short1", "string1");
+
+        OrcDataReader reader = new OrcDataReader(new File("example/data/input/input_orc_file.orc"))
+                .setColumns(columns) // Comment this line to read all columns.
```
Comment (DeleTaylor) [inline src/main/java/com/northconcepts/datapipeline/examples/orc/ReadSelectedFieldsFromAnOrcFile.java]:
```suggestion
                .setColumns(columns) // Remove this line to read all columns.
```

File: src/main/java/com/northconcepts/datapipeline/examples/orc/WriteAnOrcFile.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+import com.northconcepts.datapipeline.orc.OrcDataReader;
+import com.northconcepts.datapipeline.orc.OrcDataWriter;
+
+public class WriteAnOrcFile {
+
+    private static final File ORC_FILE = new File("example/data/output/WriteAnOrcFile.orc");
+
+    public static void main(String[] args) {
+        System.out.println("============================================================");
+        System.out.println("Write records to a parquet file");
```
Comment (DeleTaylor) [inline src/main/java/com/northconcepts/datapipeline/examples/orc/WriteAnOrcFile.java]:
```suggestion
        System.out.println("Write records to an Orc file");
```

**Review by DeleTaylor — CHANGES_REQUESTED** [review]
@gvjoshi25 we may be saying "Parquet" instead of "Orc" in some Orc examples.  Can you look into it?

File: src/main/java/com/northconcepts/datapipeline/examples/amazons3/ReadAnOrcFileFromAmazonS3UsingATemporaryFile.java (hunk 1 above)
Comment (gvjoshi25) [reply src/main/java/com/northconcepts/datapipeline/examples/amazons3/ReadAnOrcFileFromAmazonS3UsingATemporaryFile.java]: I have refactored code which is better to understand.

File: src/main/java/com/northconcepts/datapipeline/examples/amazons3/WriteAnOrcFileToAmazonS3UsingATemporaryFile.java (hunk 2 above)
Comment (gvjoshi25) [reply src/main/java/com/northconcepts/datapipeline/examples/amazons3/WriteAnOrcFileToAmazonS3UsingATemporaryFile.java]: I have refactored code which is better to understand.

File: src/main/java/com/northconcepts/datapipeline/examples/orc/ReadSelectedFieldsFromAnOrcFile.java (hunk 3 above)
Comment (gvjoshi25) [reply src/main/java/com/northconcepts/datapipeline/examples/orc/ReadSelectedFieldsFromAnOrcFile.java]: Done.

File: src/main/java/com/northconcepts/datapipeline/examples/orc/WriteAnOrcFile.java (hunk 4 above)
Comment (gvjoshi25) [reply src/main/java/com/northconcepts/datapipeline/examples/orc/WriteAnOrcFile.java]: Done.

**Comment by gvjoshi25** [comment]
> @gvjoshi25 we may be saying "Parquet" instead of "Orc" in some Orc examples. Can you look into it?

I have corrected those occurrences.

**Review by DeleTaylor — APPROVED** [review]

**Pull request merged**